tests/test_http_request.py

```diff
@@ -1147,5 +1148,57 @@ def test_latin1(self):
         self._test_request(params=(u'pas£',), encoding='latin1')
 
 
+class JSONRequestTest(RequestTest):
+    request_class = JSONRequest
+    default_method = 'POST'
+    default_headers = {b'Content-Type': [b'application/json']}
+
+    def test_body(self):
+        r1 = self.request_class(url="http://www.example.com/")
+        self.assertEqual(r1.body, b'{}')
+
+        r2 = self.request_class(url="http://www.example.com/", body=b"")
+        self.assertEqual(r2.body, b'{}')
+
+        data = {
+            'name': 'value',
+        }
+        r3 = self.request_class(url="http://www.example.com/", data=data)
+        self.assertEqual(r3.body, to_bytes(json.dumps(data)))
+
+        r4 = self.request_class(url="http://www.example.com/", body='body1', data=data)
```

Comment:
I think an exception (or at least a warning) should be printed when both `body` and `data` is used in the constructor.